apps/challenges/views.py

```diff
@@ -340,3 +357,161 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['POST'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def create_challenge_using_zip_file(request, challenge_host_team_pk):
+    """
+    Creates a challenge using a zip file.
+    """
+    try:
+        challenge_host_team =  ChallengeHostTeam.objects.get(pk=challenge_host_team_pk)
+    except:
+        response_data = {'error': 'Challenge Host Team for {} does not exist'.format(request.user)}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    serializer = ChallengeConfigurationSerializer(data=request.data, context={'request': request})
+    if serializer.is_valid():
+        uploaded_zip_file = serializer.save()
+        uploaded_zip_file_path = serializer.data['zip_configuration']
+    else:
+        response_data = serializer.errors
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
+        with transaction.atomic():
+            response = requests.get(uploaded_zip_file_path, stream=True)
+
+            # All files download and extract location.
+            BASE_LOCATION = tempfile.mkdtemp()
+
+            CHALLENGE_ZIP_DOWNLOAD_LOCATION = os.path.join('{}/zip_challenge.zip'.format(BASE_LOCATION))
+
+            if response and response.status_code == 200:
+                with open(str(CHALLENGE_ZIP_DOWNLOAD_LOCATION), 'w') as f:
```

Comment:
CHALLENGE_ZIP_DOWNLOAD_LOCATION is already a string. I think there is no need of using it :)